### PR TITLE
Refactoring `dc.datasets()` arguments

### PR DESCRIPTION
**Pull request by ilongin**
Changes to `dc.datasets()` :

- [x] renamed `object_name` to `column`
- [x] removed `column` default value of `dataset` and set default to `None` which means no top level object will be created if not specified explicitly

### CODE REVIEW

File: src/datachain/lib/dc/datasets.py [hunk 1]
```diff
@@ -102,7 +100,7 @@ def datasets(
     session: Optional[Session] = None,
     settings: Optional[dict] = None,
     in_memory: bool = False,
-    object_name: str = "dataset",
```
Comment (shcheklein) [inline src/datachain/lib/dc/datasets.py:105]: do we use object_name in some other places? should we review and change that? (for consistency)

File: src/datachain/lib/dc/datasets.py [hunk 2]
```diff
@@ -112,7 +110,8 @@ def datasets(
         session: Optional session instance. If not provided, uses default session.
         settings: Optional dictionary of settings to configure the chain.
         in_memory: If True, creates an in-memory session. Defaults to False.
-        object_name: Name of the output object in the chain. Defaults to "dataset".
+        column: Name of the output object in the chain. Defaults to None which
+            means no top level object will becreated.
```
Comment (shcheklein) [inline src/datachain/lib/dc/datasets.py:114]: typo here (please use some local linter?)

File: src/datachain/lib/dc/datasets.py [hunk 3]
```diff
@@ -112,7 +110,8 @@ def datasets(
         session: Optional session instance. If not provided, uses default session.
         settings: Optional dictionary of settings to configure the chain.
         in_memory: If True, creates an in-memory session. Defaults to False.
-        object_name: Name of the output object in the chain. Defaults to "dataset".
+        column: Name of the output object in the chain. Defaults to None which
```
Comment (shcheklein) [inline src/datachain/lib/dc/datasets.py:113]: this description (column is an object) looks weird ...  if we are renaming this, we should also change the description I think

File: src/datachain/lib/dc/datasets.py (hunk 1 above)
Comment (ilongin) [reply src/datachain/lib/dc/datasets.py:105]: Yes, we use it on multiple places. I will create separate issue for that to be clean with issues and related PRs

Comment (ilongin) [reply src/datachain/lib/dc/datasets.py:105]: https://github.com/iterative/datachain/issues/1016

**Comment by dreadatour** [comment]
We have this discussion during planning today. We now have three different options:
- `object_name`
- `column`
- `signal`

We should choose one and stick to it. Me personally I prefer `signal` rather than `column`, since we may have complex nested signals, e.g. `file: File` _signal_ which have it own _signals_ inside, like `file.name`. In case of _columns_ it will be `file__name`, but we don't really want to expose these to users. This is what I am keen to, will be happy to hear your thoughts, just wanted to discuss this here.

**Review by shcheklein — APPROVED** [review]

**Comment by ilongin** [comment]
> We have this discussion during planning today. We now have three different options:
> 
> * `object_name`
> * `column`
> * `signal`
> 
> We should choose one and stick to it. Me personally I prefer `signal` rather than `column`, since we may have complex nested signals, e.g. `file: File` _signal_ which have it own _signals_ inside, like `file.name`. In case of _columns_ it will be `file__name`, but we don't really want to expose these to users. This is what I am keen to, will be happy to hear your thoughts, just wanted to discuss this here.
> 
> **Update**: e.g. from the nearest [PR](https://github.com/iterative/datachain/pull/1020) we can see we have `SignalSchema`, `select_except_signals`, "cannot resolve signal name 'fr1.label'" error etc.

I agree this must be consistent all across the codebase. I've created related issue to do so in other parts after this PR is merged.
Regarding the exact name, I'm not 100% sure what's better. It makes sense what you said, but on the other hand `signal` is kind of a strange name to me .... not sure if its widely used in the industry for things like this? `column` is maybe more clear to the users, even though it's not mapped 1-1 to actual column in DB. 
I would like to hear other opinions as well @shcheklein @dmpetrov

**Review by dreadatour — APPROVED** [review]
❤️

**Comment by dmpetrov** [comment]
> `signal` is kind of a strange name to me .... not sure if its widely used in the industry for things like this? `column` is maybe more clear to the users

Agree. We should use common term Column instead of Signal and other terms.

**Review by amritghimire — APPROVED** [review]
LGTM. Also. would be great if we could add a test where the column is None, i.e default arg.

**Comment by amritghimire** [comment]
> > LGTM. Also. would be great if we could add a test where the column is None, i.e default arg.
> 
> @amritghimire the test is already there [#1012 (files)](https://github.com/iterative/datachain/pull/1012/files#diff-7fa3fd7accc76fc7bbc0c5e9b696be6e54cdcc5126b8c1f682b0f75cba1f95baR304-R308) Did you mean on something different or?

Yes, sorry did not notice that
